MobileSyncExplorerSwift/MobileSyncExplorerSwift/NotificationListModel.swift

```diff
@@ -0,0 +1,187 @@
+
+    private enum CodingKeys: String, CodingKey {
+        case id = "id"
+        case title = "messageTitle"
+        case body = "messageBody"
+        case read = "read"
+        case seen = "seen"
+        case imageURL = "image"
+        case targetId = "target"
+        case lastModified = "lastModified"
+    }
+
+    private static let dateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+        dateFormatter.dateFormat = "MMMM d"
+        return dateFormatter
+    }()
+
+    required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.id = try container.decode(String.self, forKey: .id)
+        self.title = try container.decode(String.self, forKey: .title)
+        self.body = try container.decode(String.self, forKey: .body)
+        self.imageURL = try container.decode(String.self, forKey: .imageURL)
+        self.targetId = try container.decode(String.self, forKey: .targetId)
+        self.read = try container.decode(Bool.self, forKey: .read)
+        self.seen = try container.decode(Bool.self, forKey: .seen)
+
+        let dateString = try container.decode(String.self, forKey: .lastModified)
+        if let date = FormatUtils.getDateFromIsoDateString(dateString) {
+            self.prettyDate = Notification.dateFormatter.string(from: date)
+        } else {
+            self.prettyDate = ""
+        }
+        fetchPhoto()
+    }
+
+    func fetchPhoto() {
+        let request = RestRequest.init(method: .GET, path: imageURL, queryParams: nil)
+        RestClient.shared.publisher(for: request)
+            .receive(on: RunLoop.main)
+            .sink(receiveCompletion: { completion in
+                if case .failure(let error) = completion {
+                    SalesforceLogger.e(Notification.self, message: "Error getting image: \(error)")
+                }
+            }, receiveValue: { [weak self] response in
+                if let image = UIImage(data: response.asData()) {
+                    self?.image = image
+                } else {
+                    SalesforceLogger.e(Notification.self, message: "Unable create UIImage from response")
+                }
+            })
+            .store(in: &cancellableSet)
+    }
+}
+
+struct NotificationResponse: Decodable {
+    var notifications: [Notification]
+}
+
+class NotificationListModel: ObservableObject {
+    @Published var notifications: [Notification] = []
+    @Published var image: UIImage?
+    private var cancellableSet: Set<AnyCancellable> = []
+
+    func unreadCount() -> Int {
+        return notifications.filter { !$0.read }.count
+    }
+
+    func fetchNotifications() {
+        let builder = FetchNotificationsRequestBuilder.init()
+        builder.setBefore(Date())
+        builder.setSize(5)
+        let request = builder.buildFetchNotificationsRequest(SFRestDefaultAPIVersion)
```

Comment:
Is there any setup needed on the server to produce those notifications?

Reply:
Yes, creating a custom notification type and then tying it to the connected app. I'd been thinking about something we could push to the org but it'd still be dependent on the connected app. Sue and I were also talking about outlining the other steps in the post she's working on

Reply:
👍